### ISSUE TITLE
Plugin not generating docgen for flow types

### ISSUE THREAD

**Issue by PDarkTemplar** [storybooks/babel-plugin-react-docgen#39]
Current docgen has issue with flow syntax like this:

```
type Colors =
    |  'text'
    | 'primary'
    | 'accent'
    | 'error'
    | 'success'
    | 'warning'
    | 'info'
    | 'additional'
    | 'minor';
```

which is perfectly fine syntax, but it is crashing on first pipe. [#236](https://github.com/reactjs/react-docgen/issues/236). With beta version it is working as expected. Can i somehow configure this plugin to use beta version of docgen with storybook? Right now i can use only workaround without using first pipe.

**Comment by danielduan** [comment]
You can do some npm or yarn tricks to have it upgrade a subdependency. I don't feel too comfortable putting beta packages in a release version.

Even though I can publish a beta for this plugin, you still won't be able to use it in Storybook..

**Comment by good-idea** [comment]
I'm having a somewhat similar issue. It's hard to pin down if it's something related to this plugin, or with **storybook/addon-info**, where I'm implementing it.

My component looks like this:

```
// @flow
import React from 'react'
import styled from 'styled-components'

import type { User } from 'Types'
import { ResponsiveImage } from 'Components/Media'
import { avatarWrapperStyles } from './styles'

const AvatarWrapper = styled.div`
	${avatarWrapperStyles};
`

/**
 * Avatar
 */

type Props = {
	size?: 'large' | 'medium' | 'small' | 'mini',
	user: User,
}

const Avatar = (props: Props) => {
	const { avatar } = props.user
	return (
		<AvatarWrapper size={props.size}>
			<ResponsiveImage ratio={1} url={avatar.url} srcSet={avatar.crops.find((c) => c.name === 'square').sizes} />
		</AvatarWrapper>
	)
}

Avatar.defaultProps = {
	size: 'small',
}

export default Avatar
```

The docs are outputting this table:
![image](https://user-images.githubusercontent.com/11514928/35416676-dd3a1cde-01de-11e8-9dfb-3e9d34b646d2.png)

It looks like it's only picking up the info from `Avatar.defaultProps` but not the flow-typed `Props`.

Output of: `./node_modules/.bin/react-docgen src/components/User/Avatar.js --pretty` ->

```
{
  "description": "",
  "methods": [],
  "props": {
    "size": {
      "flowType": {
        "name": "union",
        "raw": "'large' | 'medium' | 'small' | 'mini'",
        "elements": [
          {
            "name": "literal",
            "value": "'large'"
          },
          {
            "name": "literal",
            "value": "'medium'"
          },
          {
            "name": "literal",
            "value": "'small'"
          },
          {
            "name": "literal",
            "value": "'mini'"
          }
        ]
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "'small'",
        "computed": false
      }
    },
    "user": {
      "flowType": {
        "name": "User"
      },
      "required": true,
      "description": ""
    }
  }
}
```

I'm not sure where to go from here. Is this maybe an issue (or configuration problem) with the addon, and I should be posting there? Or, maybe a react-docgen 3.0 thing I just need to be patient for?

**Comment by Hypnosphi** [comment]
> you still won't be able to use it in Storybook

We can actually support supplying your own instance of babel-plugin-react-docgen in custom babelrc, by adding a check here: https://github.com/storybooks/storybook/blob/master/app/react/src/server/babel_config.js#L79

**Comment by danielduan** [comment]
v1.9.0 published with docgen@3.0-beta. clear your lock files, reinstall, and let me know if this doesn't fix the issue.
